src/main/java/com/rackspace/ceres/app/services/QueryService.java

```diff
@@ -101,11 +121,30 @@ public QueryService(ReactiveCqlTemplate cqlTemplate,
         .checkpoint();
   }
 
-  public Flux<QueryResult> queryDownsampled(String tenant, String metricName, Aggregator aggregator,
+  public Flux<QueryResult> queryDownsampled(String tenant, String metricKey, Aggregator aggregator,
       Duration granularity, Map<String, String> queryTags,
       Instant start, Instant end) {
+    return getMetricGroupFlux(metricKey)
+        .flatMap(
+            metricName -> {
+              String metricNameWithGroup = metricKey+"_"+metricName;
```

Comment:
Hmm.. shouldn't a group be translated into queries with metricNames, I'm not sure I like this underscore thing...

Reply:
Yeah this shouldn't need any string concatenation. The result set from redis should be the metricNames to query for in Cassandra.